iquip/apps/builder.py

```diff
@@ -574,6 +578,100 @@ def scanArguments(self) -> Dict[str, Any]:
         }
 
 
+class _CenterScan(_BaseScan):
+    """Widget for center scan in _ScanEntry.
+
+    Attributes:
+        centerSpinBox: QDoubleSpinBox for center argument inside state.
+        spanSpinBox: QDoubleSpinBox for span argument inside state.
+        stepSpinBox: QDoubleSpinBox for step argument inside state.
+        randomizeCheckBox: QCheckBox for randomize argument inside state.
+    """
+
+    def __init__(
+        self,
+        procdesc: Dict[str, Any],
+        state: Dict[str, Any],
+        parent: Optional[QWidget] = None
+    ):
+        """Extended.
+
+        Args:
+            state: Each key and its value are:
+              center: The center point for the CenterScan sequence.
+              span: The length of the CenterScan sequence.
+              step: The size of step between each number in the CenterScan sequence.
+              randomize: The boolean value that decides whether to shuffle the CenterScan sequence.
+        """
+        super().__init__(procdesc, parent=parent)
+        self.centerSpinBox = QDoubleSpinBox(self)
+        self.applyProperties(self.centerSpinBox)
+        self.centerSpinBox.setValue(state["center"] / self.scale)
+        self.spanSpinBox = QDoubleSpinBox(self)
+        self.applyProperties(self.spanSpinBox)
+        self.spanSpinBox.setValue(state["span"] / self.scale)
+        self.stepSpinBox = QDoubleSpinBox(self)
+        self.applyProperties(self.stepSpinBox)
+        self.stepSpinBox.setValue(state["step"] / self.scale)
+        self.randomizeCheckBox = QCheckBox("Randomize", self)
+        self.randomizeCheckBox.setChecked(state["randomize"])
+        # layout
+        self.layout.addWidget(QLabel("center:", self), 0, 0)
+        self.layout.addWidget(self.centerSpinBox, 0, 1)
+        self.layout.addWidget(QLabel("span:", self), 1, 0)
+        self.layout.addWidget(self.spanSpinBox, 1, 1)
+        self.layout.addWidget(QLabel("step:", self), 2, 0)
+        self.layout.addWidget(self.stepSpinBox, 2, 1)
+        self.layout.addWidget(self.randomizeCheckBox, 3, 1)
+
+    def scanArguments(self) -> Dict[str, Any]:
+        """Overridden."""
+        return {
+            "ty": "CenterScan",
+            "center": self.centerSpinBox.value(),
+            "step": self.stepSpinBox.value(),
+            "span": self.spanSpinBox.value(),
+            "randomize": self.randomizeCheckBox.isChecked(),
+            "seed": None
+        }
+
+
+class _ExplicitScan(_BaseScan):
+    """Widget for explicit scan in _ScanEntry.
+
+    Attributes:
+        sequenceEdit: QLineEdit for sequence argument inside state.
+    """
+
+    def __init__(
+        self,
+        procdesc: Dict[str, Any],
+        state: Dict[str, Any],
+        parent: Optional[QWidget] = None
+    ):
+        """Extended.
+
+        Args:
+            state: A key and its value is:
+              sequence: The sequnce that describes the ExplicitScan sequence.
```

Comment:
The "sequnce" is a typo, and you already wrote "sequence" in the back, so it looks better to replace it with another word.
For example, "The list that represents the ExplicitScan sequence."